popup.js

```diff
@@ -1,18 +1,119 @@
-document.addEventListener('DOMContentLoaded', function() {
-    console.log('dom content loaded');
-    console.log(chrome);
-    console.log(chrome.tabs);
-    chrome.tabs.query({active: true, currentWindow: true}, function(tabs) {
-        console.log(tabs);
-    });
+function setup() {
+    document.addEventListener('DOMContentLoaded', async function() {
+        // fetch language data from the API and dynamically load elements into the drop-down menus.
+        const resp = await fetch('https://libretranslate.eownerdead.dedyn.io/languages');
+        const languageData =  await resp.json();
+        loadLanguages(languageData);
+
+        // setup local storage for the extension to access language features.
+        /*
+        1. Initialise the user's language preferences during translation.
+        2. Set the selected property of the drop-down menus' options accordingly.
+         */
+        const prefs = await initialiseLanguageSelection();
+        console.log('What is prefs? ' + JSON.stringify(prefs));
+        initialiseLanguagePreferences(prefs.source, prefs.target);
 
-    const downloadButton = document.querySelector('button');
-    if (downloadButton) {
-        downloadButton.addEventListener('click', function() {
-            setupDownloadButton();
+        sourceLanguageListener();
+        targetLanguageListener();
+
+        console.log('dom content loaded');
+        console.log(chrome);
+        console.log(chrome.tabs);
+        chrome.tabs.query({active: true, currentWindow: true}, function(tabs) {
+            console.log(tabs);
         });
+
+        const downloadButton = document.querySelector('button');
+        if (downloadButton) {
```

Comment:
while it is good to check if this exists as it's possible that it doesn't, we're not doing this elsewhere so it's inconsistent to do that here

in this case it's fine to just assume `document.querySelector()` got the right element, an error will just be thrown if not and we'll take it as it comes

Reply:
You're right, plus the whole extension would be (mostly) useless if the elements aren't loaded correctly, in the popup. I've removed the if-statement now.